layers/core_checks/cc_copy_blit_resolve.cpp

```diff
@@ -564,94 +557,94 @@ uint32_t GetImageHeight<VkImageToMemoryCopyEXT>(VkImageToMemoryCopyEXT data) {
 }
 template <typename HandleT, typename RegionType>
 bool CoreChecks::ValidateHeterogeneousCopyData(const HandleT handle, uint32_t regionCount, const RegionType *pRegions,
-                                               const IMAGE_STATE &image_state, const char *function, const Location &loc,
-                                               bool from_image, bool is_memory) const {
-    // Validation common to vkCmdCopyImageToBuffer, vkCmdCopyBufferToImage, vkCopyMemoryToImageEXT, and vkCopyImageToMemoryEXT
+                                               const IMAGE_STATE &image_state, const Location &loc) const {
     bool skip = false;
     const bool is_2 = loc.function == Func::vkCmdCopyBufferToImage2 || loc.function == Func::vkCmdCopyBufferToImage2KHR;
+    const bool from_image = loc.function == Func::vkCmdCopyImageToBuffer || loc.function == Func::vkCmdCopyImageToBuffer2 ||
+                            loc.function == Func::vkCmdCopyImageToBuffer2KHR || loc.function == Func::vkCopyImageToMemoryEXT;
```

Comment:
Suggestion, (and self promotion): I wrote a utility function that would allow to write this logic like so:
```suggestion
    const bool from_image = IsValueIn(loc.function, {Func::vkCmdCopyImageToBuffer, Func::vkCmdCopyImageToBuffer2, Func::vkCmdCopyImageToBuffer2KHR, Func::vkCopyImageToMemoryEXT});
```
IMO it is a bit more readable to use this function when the list of possible values is >= 3

Reply:
agree, thanks for the promoting!